pkg/transmitter/util.go

```diff
@@ -134,15 +134,7 @@ func generateTransmitterFrom(ctx context.Context, rargs types.RelayArgs, ethKeys
 		checker.CheckerType = evmtxmgr.TransmitCheckerTypeSimulate
 	}
 
-	gasLimit := chain.Config().EVM().GasEstimator().LimitDefault()
-	ocr2Limit := chain.Config().EVM().GasEstimator().LimitJobType().OCR2()
-	if ocr2Limit != nil {
-		gasLimit = uint64(*ocr2Limit)
-	}
-	if opts.PluginGasLimit != nil {
-		gasLimit = uint64(*opts.PluginGasLimit)
-	}
-
+	gasLimit := getGasLimitFrom(chain, opts, relayConfig)
```

Comment:
You can use `cmp.Or` to select the first non-nil value, and then the func maybe isn't necessary:
```suggestion
	gasLimit := chain.Config().EVM().GasEstimator().LimitDefault()
	ocr2Limit := chain.Config().EVM().GasEstimator().LimitJobType().OCR2()
	if gl := cmp.Or(ocr2Limit, opts.PluginGasLimit, relayConfig.GasLimit); gl != nil {
		gasLimit = uint64(*gl)
	}
```

Reply:
Nice idea! A downside is that the func helps with testability. 
I'm open to both, any preference?

Reply:
If you want to keep it, then I think it should be simplified. e.g. it doesn't need the whole `legacyevm.Chain` - it just needs the gas estimator interface.

Reply:
Don't need the whole RelayConfig either

Reply:
Updated!